src/shared/libcrypt-util.c

```diff
@@ -90,6 +90,32 @@ int make_salt(char **ret) {
 #endif
 }
 
+#if !HAVE_CRYPT_RA
+/* Provide a poor man's fallback that uses a fixed size buffer. */
+
+static char* systemd_crypt_ra(const char *phrase, const char *setting, void **data, int *size) {
+        _cleanup_(erase_and_freep) void *cd = NULL;
+
+        if (!data || !*data) {
```

Comment:
hmm, the non-systemd version of crypt_ra() doesn't support `data == NULL` afaics. Either this function shouldn't either, or we provide a wrapper of some form, maybe `safe_crypt_ra()` that adds this also for the libxcrypt provided version